Openvswitch db

markleehamilton opened the pull request:
Module for changing values in openvswitch tables.

bcoca commented on network/openvswitch_db.py:
```diff
@@ -0,0 +1,129 @@
+
+    (_, output, _) = cmd_run(module, cmd, False)
+    if module.params['value'] not in output:
+        fmt = "ovs-vsctl -t %(timeout)s set %(table)s %(record)s " \
```
use modules find_bin function instead of assuming this is in PATH

markleehamilton commented:
bcoca, I updated the module to use get_bin_path.

bcoca commented on network/openvswitch_db.py:
```diff
@@ -0,0 +1,135 @@
+            - How long to wait for ovs-vswitchd to respond
+"""
+
+DOCUMENTATION = __doc__
```
not sure why a 2nd 'DOCUMENTATION is here, but this will break the docs parser

bcoca commented on network/openvswitch_db.py:
```diff
@@ -0,0 +1,135 @@
+
+def cmd_run(module, cmd, check_rc=True):
+    """ Log and run ovs-vsctl command. """
+    syslog.syslog(syslog.LOG_NOTICE, cmd)
```
you should not invoke syslog, the module already does this with pertinent information

bcoca commented on network/openvswitch_db.py:
```diff
@@ -0,0 +1,135 @@
+        (rtc, _, err) = cmd_run(module, cmd)
+        if rtc != 0:
+            module.fail_json(msg=err)
+        changed = True
```
this seems to always return 'changed' unless there is an error, is there no way to check if the values already exist as expected?

markleehamilton replied:
Line 98 does the check to see if the value is already set. The module can return not changed.

markleehamilton commented:
I removed the extra syslog in cmd_run method. I find the way I posted to syslog easier to read but I get your concern. I also removed the second DOCUMENTATION variable.

markleehamilton commented:
bcoca,
 I addressed all of your concerns.
